vivarium/framework/configuration.py

```diff
@@ -0,0 +1,52 @@
+import os.path
+
+from vivarium import VivariumError
+from vivarium.config_tree import ConfigTree
+
+from .plugins import DEFAULT_PLUGINS
+
+
+class ConfigurationError(VivariumError):
+    """Error raised when invalid configuration is received."""
+    pass
+
+
+def build_model_specification(model_specification_file_path: str) -> ConfigTree:
+    _validate_model_specification_file(model_specification_file_path)
+
+    model_specification = _get_default_specification()
+
+    user_config_path = os.path.expanduser('~/vivarium.yaml')
+    if os.path.exists(user_config_path):
+        model_specification.configuration.update(user_config_path, layer='component_configs')
+
+    model_specification.update(model_specification_file_path, layer='model_override')
+
+    return model_specification
+
+
+def build_simulation_configuration() -> ConfigTree:
+    config = _get_default_specification().configuration
+    user_config_path = os.path.expanduser('~/vivarium.yaml')
+    if os.path.exists(user_config_path):
+        config.update(user_config_path, layer='component_configs')
+    return config
+
+
+def _validate_model_specification_file(model_specification_file: str) -> None:
+    if not os.path.isfile(model_specification_file):
+        raise ConfigurationError('If you provide a model specification file, it must be a file. '
+                                 f'You provided {model_specification_file}')
+
+    extension = model_specification_file.split('.')[-1]
+    if extension not in ['yaml', 'yml']:
```

Comment:
There is a more complete validation in utilities.py. Might make sense to call that here.